sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -0,0 +1,1126 @@
+            if (responseItem.hasNext()) {
+                return Mono.just(new SimpleResponse<>(response, responseItem.next()));
+            }
+            return monoError(logger, new RuntimeException("Unable to retrieve language for the provided text."));
+        });
+    }
+
+    /**
+     * Returns the detected language for a batch of input.
+     *
+     * @param inputs The list of texts to be analyzed.
+     * @return A {@link Mono} containing the {@link DocumentResultCollection batch} of the
+     * {@link DetectLanguageResult detected languages}.
+     * @throws NullPointerException if {@code inputs} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DocumentResultCollection<DetectLanguageResult>> detectLanguages(List<String> inputs) {
+        return detectLanguagesWithResponse(inputs, null).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Returns the detected language for a batch of input with the provided country hint.
+     *
+     * @param inputs The list of texts to be analyzed.
+     * @param countryHint A country hint for the entire batch. Accepts two letter country codes specified by ISO 3166-1
+     * alpha-2. Defaults to "US" if not specified.
+     * @return A {@link Response} of {@link Mono} containing the {@link DocumentResultCollection batch} of the
+     * {@link DetectLanguageResult detected languages}.
+     * @throws NullPointerException if {@code inputs} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<DocumentResultCollection<DetectLanguageResult>>> detectLanguagesWithResponse(
+        List<String> inputs, String countryHint) {
+        try {
+            return withContext(context -> detectLanguagesWithResponse(inputs, countryHint, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<DocumentResultCollection<DetectLanguageResult>>> detectLanguagesWithResponse(List<String> inputs,
+                                                                                               String countryHint,
+                                                                                               Context context) {
+        List<DetectLanguageInput> languageInputs = getLanguageInputList(inputs, countryHint);
+        return detectBatchLanguagesWithResponse(languageInputs, null, context);
+    }
+
+    /**
+     * Helper method to convert text list input to LanguageInput.
+     *
+     * @param inputs the list of user provided texts.
+     * @param countryHint the countryHint provided by user for texts.
+     *
+     * @return the LanguageInput list objects to provide the service.
+     */
+    private static List<DetectLanguageInput> getLanguageInputList(List<String> inputs, String countryHint) {
+        List<DetectLanguageInput> languageInputs = new ArrayList<>();
```

Comment:
There is a lot of duplication with this mapping to an object. You can create a function for this.
```java
    private foo() {
        List<String> inputs = Arrays.asList("bar", "foo");
        String countryCode = "something";

        List<DetectLanguageInput> detectLanguageInputs = mapByIndex(inputs, (index, value) -> new DetectLanguageInput(index, value, countryCode));
    }

    private static <T> List<T> mapByIndex(List<String> inputs, BiFunction<String, String, T> mappingFunction) {
        return IntStream.range(0, inputs.size())
            .mapToObj(index -> mappingFunction.apply(String.valueOf(index), inputs.get(index)))
            .collect(Collectors.toList());
    }
```